core/services/ping/portwatcher.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import logging
+from typing import Callable, Dict, List, Optional, Set
+from warnings import warn
+
+import serial.tools.list_ports
+from serial.tools.list_ports_linux import SysFS
+
+from pingutils import PingDeviceDescriptor
+
+MAX_ATTEMPTS = 10
+
+
+def pingDevicesFilter(port: SysFS) -> bool:
+    """Accepts ports with known chips in the Ping1D and Ping360. Internal ports
+    Are handled by Ardupilot and show None for interface, so we ignore those too.
+    """
+    if not port.interface:
+        return False
+    return "FT231X" in port.interface or "FT230" in port.interface
```

Comment:
That may be a bit problematic since the use can use different serial-usb chips, the user can use a ch341 or anything like it.  Maybe you want to use a blacklist over a whitelist ? It makes more sense.

Reply:
maybe a user-configurable black/whitelist?
The issue with a blacklist, is that the user will need know that the probing will happen, and that his device will need to handle it or he'll need to blacklist it in the ping(or higher level in core) blacklist.

Reply:
But the same thing happens to the whitelist, like an user with an arduino that contains a ftdi. 
From what I saw already is that users may not use FT230/FT231X, there are people also that uses Ping360 with an RS485 to USB adapter.

Should we let it open for users to configure ? If so, should we have an REST API for that ? Should we support both white and blacklist or only one ?

Reply:
What I have in mind right now is to have one blacklist for the whole system.
This way the user can tell us "please don't touch this device". and we avoid it in all our services.
we would need to expose this blacklist eventually (maybe on a settings editor?).

For our own devices, I don't expect we'll have trouble making sure they handle being probed.

My concern is probing badly-written arduino peripherals done by our users and putting them in a state where the user's service is unable to work properly.

Reply:
After chatting to @patrickelectric :
Try to udev rules to link all not-blacklisted devices to /dev/companion. Our drivers should test the devices in /dev/companion only.

Reply:
I opened https://github.com/bluerobotics/companion-docker/issues/237